### PR TITLE
PR: Enable filtering in multiple columns in the shortcuts table (Preferences)

### LINKED ISSUE #13144: I'm not able to search by context in the shortcuts table
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
The find widget inside the shortcuts table is not searching by context, just by the name of the shortcut

### What steps reproduce the problem?

1. Open the preference menu
2. Go to Keyboard shortcuts 
3. Try to find all the shortcuts in a given context

### What is the expected output? What do you see instead?
See the shortcuts of the typed context

### PR DESCRIPTION
**Pull request by steff456**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Add support for filering in multiple columns in the table of shortcuts. Now the user can search the shortcut by context and name.

find

### Issue(s) Resolved

Fixes #13144

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Steff456

### CODE REVIEW

**Comment by ccordoba12** [comment]
@steff456, please rebase to get the fix pushed yesterday to our tests.

File: spyder/preferences/tests/test_shorcuts.py [hunk 1]
```diff
@@ -114,6 +114,25 @@ def test_shortcuts_filtering(shortcut_table):
     assert shortcut_table.model().rowCount() == row_count
 
 
+def test_shortcut_filtering_context(shortcut_table):
+    """Test the multifiltering by context and name in the table."""
```
Comment (ccordoba12) [inline spyder/preferences/tests/test_shorcuts.py:118]:
```suggestion
    """Test multifiltering by context and name in the table."""
```

File: spyder/widgets/helperwidgets.py [hunk 2]
```diff
@@ -375,6 +375,65 @@ def filterAcceptsRow(self, row_num, parent):
...
+            self.pattern = get_search_regex(text)
+            if self.pattern and text:
+                self._parent.setSortingEnabled(False)
+            else:
+                self._parent.setSortingEnabled(True)
+            self.filters[key] = self.pattern
+            self.invalidateFilter()
+
+    def clearFilter(self, column):
+        """Clear the filter of given column."""
```
Comment (ccordoba12) [inline spyder/widgets/helperwidgets.py:405]:
```suggestion
        """Clear the filter of the given column."""
```

File: spyder/widgets/helperwidgets.py [hunk 3]
```diff
@@ -375,6 +375,65 @@ def filterAcceptsRow(self, row_num, parent):
...
+    def clearFilters(self):
+        """Clear all the filters."""
+        self.filters = {}
+        self.invalidateFilter()
+
+    def filterAcceptsRow(self, row_num, parent):
+        """Qt override.
+
+        Reimplemented from base class to allow the use of multiple filtering.
+        """
```
Comment (ccordoba12) [inline spyder/widgets/helperwidgets.py:418]:
```suggestion
        """
        Qt override.

        Reimplemented to allow filtering in multiple columns.
        """
```

**Comment by ccordoba12** [comment]
@dalthviz, please take a loot at @steff456's code because you're the expert on proxy models among us.

File: spyder/widgets/helperwidgets.py [hunk 4]
```diff
@@ -375,6 +375,65 @@ def filterAcceptsRow(self, row_num, parent):
             return True
 
 
+class MultipleSortFilterProxy(QSortFilterProxyModel):
```
Comment (dalthviz) [inline spyder/widgets/helperwidgets.py:378]: Maybe this should be named `ShortcutsSortFilterProxy` and should go in `spyder/preferences/shortcuts.py`?

File: spyder/preferences/tests/test_shorcuts.py [hunk 5]
```diff
@@ -114,6 +114,25 @@ def test_shortcuts_filtering(shortcut_table):
...
 
 
+def test_shortcut_filtering_context(shortcut_table):
+    """Test multifiltering by context and name in the table."""
+    # Verify that the model is filtering by two columns
+    assert len(shortcut_table.model().filters) == 2
+    # Filter by "console"
+    shortcut_table.finder = FilterTextMock('console')
+    shortcut_table.set_regex()
+    # Verify the number of entries after the regex are 10
```
Comment (dalthviz) [inline spyder/preferences/tests/test_shorcuts.py:124]: Should the comment above say 7 instead of 10?

File: spyder/widgets/helperwidgets.py (hunk 4 above)
Comment (steff456) [reply spyder/widgets/helperwidgets.py:378]: I put it in here just in case someday we need a multiple sort filter in other place, should I move it?

**Comment by ccordoba12** [comment]
@dalthviz, please approve this one if you consider it's ready. Then I'll merge it.

**Pull request merged**